demos/python_demos/forward_tacotron_wavernn_demo/models/mel2wave_ie.py

```diff
@@ -0,0 +1,218 @@
+        self.rnn_width = self.rnn_net.inputs['x'].shape[1]
+
+    def get_rnn_init_states(self, b_size=1, rnn_dims=328):
+        h1 = np.zeros((b_size, rnn_dims), dtype=float)
+        h2 = np.zeros((b_size, rnn_dims), dtype=float)
+        x = np.zeros((b_size, 1), dtype=float)
+        return h1, h2, x
+
+    def forward_fixed_shape(self, mels):
+        upsampled_mels, aux = self.forward_upsample(mels)
+        upsampled_mels = fold_with_overlap(upsampled_mels, self.target, self.overlap)
+        aux = fold_with_overlap(aux, self.target, self.overlap)
+        audio = self.forward_rnn(mels, upsampled_mels, aux)
+
+        return audio
+
+    def forward(self, mels):
+        n_parts = mels.shape[1] // self.mel_len + 1 if mels.shape[1] % self.mel_len >0 else mels.shape[1] // self.mel_len
+        upsampled_mels = []
+        aux = []
+        last_padding = 0
+        for i in range(n_parts):
+            i_start = i * self.mel_len
+            i_end = i_start + self.mel_len
+            if i_end > mels.shape[1]:
+                last_padding = i_end - mels.shape[1]
+                mel = np.pad(mels[:, i_start:mels.shape[1], :], ((0, 0), (0, last_padding), (0, 0)), 'constant', constant_values=0)
+            else:
+                mel = mels[:,i_start:i_end, :]
+
+            upsampled_mels_b, aux_b = self.forward_upsample(mel)
+            upsampled_mels.append(upsampled_mels_b)
+            aux.append(aux_b)
+        if len(aux) > 1:
+            upsampled_mels = np.concatenate(upsampled_mels,axis=1)
+            aux = np.concatenate(aux, axis=1)
+        else:
+            upsampled_mels = upsampled_mels[0]
+            aux = aux[0]
+        if last_padding > 0:
+            upsampled_mels = upsampled_mels[:,:-last_padding*self.hop_length,:]
+            aux = aux[:,:-last_padding*self.hop_length,:]
+
+        upsampled_mels, (_, self.dynamic_overlap) = fold_with_overlap(upsampled_mels, self.target, self.overlap)
+        aux, _ = fold_with_overlap(aux, self.target, self.overlap)
+
+        audio = self.forward_rnn(mels, upsampled_mels, aux)
+
+        return audio
+
+    def forward_upsample(self, mels):
+        mels = pad_tensor(mels, pad=self.pad)
+
+        out = self.upsample_exec.infer(inputs={"mels": mels})
+        upsample_mels, aux = out["upsample_mels"][:,self.indent:-self.indent,:], out["aux"]
+        return upsample_mels, aux
+
+    def forward_upsample_one_iter(self, inputs):
+        out = self.upsample_exec.infer(inputs=inputs)
+        return out
+
+    def forward_rnn(self, mels, upsampled_mels, aux):
+        wave_len = (mels.shape[1] - 1) * self.hop_length
+
+        d = aux.shape[2] // 4
+        aux_split = [aux[:, :, d * i:d * (i + 1)] for i in range(4)]
+
+        b_size, seq_len, _ = upsampled_mels.shape
+
+        if not b_size in self.batch_sizes:
+            raise Exception('Incorrect batch size {0}. Correct should be 2 ** something'.format(b_size))
```

Comment:
You should check it right after net loaded.

Reply:
Batch size changed dynamically for different inputs, time axes is not static for this task. So it is impossible to check it after net loading. I load bunch of networks with different batch sizes to speedup computation and reshape input data batch size to near power of two.